src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -804,7 +804,6 @@ private bool WriteLocalFileHeader(bool isEmptyFile)
                 CompressionMethod = CompressionMethodValues.Stored;
                 compressedSizeTruncated = 0;
                 uncompressedSizeTruncated = 0;
-                Debug.Assert(_compressedSize == 0);
```

Comment:
Why do you need to remove it? seems you still setting this to 0 if it's empty

Reply:
Because I was setting it to 0 after calling this method. Changed it back now.